patients/service.go

```diff
@@ -240,6 +269,61 @@ func (s *service) enforceMrnSettings(ctx context.Context, clinicId string, exist
 	return nil
 }
 
+func (s *service) enforcePatientCountSettings(ctx context.Context, clinicId string, patient *Patient) error {
+
+	// Allow non-custodial patients no matter what
+	if !patient.IsCustodial() {
+		return nil
+	}
+
+	// Get any clinic patient count settings, if none found, then no limits, so allow
+	patientCountSettings, err := s.clinics.GetPatientCountSettings(ctx, clinicId)
+	if err != nil || patientCountSettings == nil {
+		return err
+	}
+
+	// If no patient count hard limit setting, then allow
+	if patientCountSettings.HardLimit == nil {
+		return nil
+	}
+
+	// If outside start date and end date, if specified, then allow
+	now := time.Now()
+	if patientCountSettings.HardLimit.StartDate != nil && now.Before(*patientCountSettings.HardLimit.StartDate) {
+		return nil
+	} else if patientCountSettings.HardLimit.EndDate != nil && now.After(*patientCountSettings.HardLimit.EndDate) {
+		return nil
```

Comment:
Shouldn't we prevent adding patients after the current plan expires, instead of allowing it?

Reply:
The hard limit only applies if the current time (`now`) is *within* the start and end dates, if specified. So, in this case, if the `EndDate` is specified and `now` is after the `EndDate`, then the limit is NOT applied, so returns `nil` here.

Reply:
Right, that was my understanding as well. However, I don't understand why would we want to have an end date for the hard limit.

From the ticket description it seems that we should capture a start and end dates for the current subscription of a clinic. We can't use the hard limit start and end dates for this, because we would be relaxing the restrictions when the plan expires.

Reply:
Just wanted to make sure the end date won't be used as the plan end date, because I don't think we should relax that patient count limit when the plan expires.

Reply:
Yes, I agree. 👍